src/renderer/store/editor.js

```diff
@@ -394,6 +394,8 @@ const actions = {
     if (cursor) commit('SET_CURSOR', cursor)
     // set history
     if (history) commit('SET_HISTORY', history)
+    // if new markdown and old markdown are empty, no need to save or change status
+    if (!/\S/.test(markdown) && !/\S/.test(oldMarkdown)) return
```

Comment:
I don't know why you check for white-spaces? Does the following not catch it the empty case too?

```js
L400 if (markdown !== oldMarkdown) {
```

Or do you want catch the `\n` case which is automatically added sometimes? Then I would check for one new-line only.